src/validator/data_model.py

```
@@ -138,4 +138,71 @@ def validate_part_of_total_floor_area(cls, v):
+    vertices: List[dict[str, float]] = Field(..., alias="Vertices", description="List of vertices defining the surface")
+
+    @field_validator('name', 'construction_name', 'zone_name')
+    def validate_non_empty(cls, v):
+        if not v:
+            raise ValueError("This field must not be empty.")
+        return v
+    @field_validator('surface_type')
+    def validate_surface_type(cls, v):
+        valid_types = {"Wall", "Roof", "Floor", "Window", "Door", "Ceiling", "AirWall", "UndergroundWall", "UndergroundSlab"}
+        if v not in valid_types:
+            raise ValueError(f"Surface Type must be one of {valid_types}.")
```

Ruff (0.13.2)

Avoid specifying long messages outside the exception class

(TRY003)